sycl/doc/extensions/proposed/sycl_ext_oneapi_work_group_static.asciidoc

```diff
@@ -146,73 +143,89 @@ specified explicitly.
-allocation in device local memory. For example, two instances declared
-as `work_group_static<int[]>` and `work_group_static<float[]>` will be
-associated with the same shared allocation.
-
-If the total amount of device local memory requested (i.e., the sum of
-all memory requested by `local_accessor`, `group_local_memory`,
-`group_local_memory_for_overwrite` and `work_group_static`) exceeds a device's
-local memory capacity (as reported by `local_mem_size`) then the implementation
-must throw a synchronous `exception` with the `errc::memory_allocation` error
-code from the kernel invocation command (e.g. `parallel_for`).
-
 [source,c++]
 ----
-operator T&() const noexcept;
+operator T&() noexcept;
 ----
 _Returns_: A reference to the object stored in the device local memory
 associated with this instance of `work_group_static`.
 
 [source,c++]
 ----
-const work_group_static<T>& operator=(const T& value) const noexcept;
+work_group_static<T>& operator=(const T& value) noexcept;
 ----
-_Constraints_: Available only if `std::is_array_v<T>>` is false.
+_Constraints_: Available only if `std::is_array_v<T>` is false.
 
 _Effects_: Replaces the value referenced by `*ptr` with `value`.
 
 _Returns_: A reference to this instance of `work_group_static`.
 
 [source,c++]
 ----
-T* operator&() const noexcept;
+T* operator&() noexcept;
 ----
 _Returns_: A pointer to the device local memory associated with this
 instance of `work_group_static` (i.e., `ptr`).
 
+=== `get_dynamic_work_group_memory` function
+
+The `get_dynamic_work_group_memory` function provides access
+to a dynamically allocated buffer in the device local memory.
+
+[source,c++]
+----
+template <typename T>
+sycl::multi_ptr<T, access::address_space::local_space,
+                           access::decorated::no>
+get_dynamic_work_group_memory()
+----
+_Constraints_: `T` must be trivially constructible and trivially destructible.
+
+_Returns_: A pointer to a dynamically allocated buffer
+           in the device local memory.
+
+The size of the allocation is unknown at compile-time,
+and must be communicated to the SYCL implementation via the
+`work_group_static_memory_size` property. Every call to
+`get_dynamic_work_group_memory` returns the same allocation
+in device local memory, regardless of `T`. For example, two call declared
+as `get_dynamic_work_group_memory<int>` and
+`get_dynamic_work_group_memory<float>` will be associated with the same shared allocation.
+
+If the total amount of device local memory requested (i.e., the sum of
+all memory requested by `local_accessor`, `group_local_memory`,
+`group_local_memory_for_overwrite` and `work_group_static`) exceeds a device's
+local memory capacity (as reported by `local_mem_size`) then the implementation
+must throw a synchronous `exception` with the `errc::memory_allocation` error
+code from the kernel invocation command (e.g. `parallel_for`).
+
 
 ==== Kernel properties
```

Comment:
```suggestion
=== Kernel properties
```

Otherwise, this is a sub-section under the description of `get_dynamic_work_group_memory`, which makes no sense.